### PR TITLE
Feat/generate queries

### PR DESCRIPTION
**Pull request by Okm165**

### CODE REVIEW

File: src/queries.cairo [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        }
+
+        i += 1;
+    };
+
+    result
+}
+
+
+#[test]
```
Comment (tiagofneto) [inline src/queries.cairo:85]: I would add all the tests on a separate file, just to follow the standard we have been following. If you really prefer to have them on this file, I would place them inside a `mod` section with `#[cfg(test)]`

File: src/common/array_split.cairo
Comment (tiagofneto) [inline src/common/array_split.cairo:1]: We are starting to have a lot of array utils. I'm wondering if it wouldn't be more worth it to define them all in the same file and some trait / implementation

Comment (Okm165) [reply src/common/array_split.cairo:1]: let's do it when we do folder structure refactor to avoid conflicts

File: src/queries.cairo (hunk 1 above)
Comment (Okm165) [reply src/queries.cairo:85]: fixed

**Review by tiagofneto — APPROVED** [review]
lgtm

File: src/queries.cairo (hunk 1 above)
Comment (fmkra) [reply src/queries.cairo]: Now for every "module" we have a separate directory and tests are in separate file (even for channel which has just a single implementation file and one test file), so I would opt for moving tests to separate file.

Comment (Okm165) [reply src/queries.cairo]: i separated it into proper module

**Review by fmkra — APPROVED** [review]
lgtm

**Pull request merged**